planner/core/memtable_predicate_extractor.go

```diff
@@ -184,6 +217,147 @@ func (helper extractHelper) extractAddressAndTypeCols(schema *expression.Schema,
 	return
 }
 
+// extracts the string pattern column, e.g:
+// SELECT * FROM t WHERE c LIKE '%a%'
+// SELECT * FROM t WHERE c LIKE '%a%' AND c REGEXP '.*xxx.*'
+func (helper extractHelper) extractLikePatternCol(
+	schema *expression.Schema,
+	names []*types.FieldName,
+	predicates []expression.Expression,
+	extractColName string,
+) (
+	remained []expression.Expression,
+	patterns []string,
+) {
+	remained = make([]expression.Expression, 0, len(predicates))
+	extractCols := make(map[int64]*types.FieldName)
+	for i, name := range names {
+		if name.ColName.L == extractColName {
+			extractCols[schema.Columns[i].UniqueID] = name
+		}
+	}
+
+	if len(extractCols) == 0 {
+		return predicates, nil
+	}
+
+	// We use a string array to save multiple patterns because the Golang and Rust don't
+	// support perl-like CNF regular expression: (?=expr1)(?=expr2).
+	// e.g:
+	// SELECT * FROM t WHERE c LIKE '%a%' AND c LIKE '%b%' AND c REGEXP 'gc.*[0-9]{10,20}'
+	for _, expr := range predicates {
+		fn, ok := expr.(*expression.ScalarFunction)
+		if !ok {
+			remained = append(remained, expr)
+			continue
+		}
+		var colName string
+		var datums []types.Datum
+
+		switch fn.FuncName.L {
+		case ast.GT, ast.GE, ast.LT, ast.LE, ast.EQ, ast.Like, ast.Regexp:
+			colName, datums = helper.extractColBinaryOpConsExpr(extractCols, fn)
+		}
+		if colName == extractColName {
+			switch fn.FuncName.L {
+			case ast.EQ:
+				patterns = append(patterns, "^"+regexp.QuoteMeta(datums[0].GetString())+"$")
+			case ast.Like:
+				patterns = append(patterns, stringutil.CompileLike2Regexp(datums[0].GetString()))
+			case ast.Regexp:
+				patterns = append(patterns, datums[0].GetString())
+			default:
+				remained = append(remained, expr)
+			}
+		} else {
+			remained = append(remained, expr)
+		}
+	}
+	return
+}
+
+// extracts the time range column, e.g:
+// SELECT * FROM t WHERE time='2019-10-10 10:10:10'
+// SELECT * FROM t WHERE time>'2019-10-10 10:10:10' AND time<'2019-10-11 10:10:10'
```

Comment:
So you don't deal with (time > ... and time < ...) or (time > ... and time < ...)?

Reply:
Yes, to simplify the implementation of the diagnostics service search log interface, it can just accept only one range of time (pick a group of the log files in the time range). So the filtering will be executed in TiDB via Selection executor.

Reply:
Generally, we use the cluster level log search function to locate the problem usually only one range of time.